File: PB - More Exercises Python/7.0 Nested Loops - More Exercises/05. Challenge the Wedding.py
men, women, table = [int(input()) for _ in range(3)]
visited = []
for m in range(1, men + 1):
    for w in range(1, women + 1):
        if table != len(visited):
            visited.append(f'{m} <-> {w}')
        else:
            break

[print(f'({x})', end=' ') for x in visited]


'''
2
2
6
'''

File: PB - More Exercises GO/7.0 Nested Loops - More Exercises/05. Challenge the Wedding.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	var men, women, table int
	fmt.Scan(&men, &women, &table)
	var visitedList []string

	for m := 1; m <= men; m++ {
		for w := 1; w <= women; w++ {
			if table != len(visitedList) {
				visitedList = append(visitedList, fmt.Sprintf("(%v <-> %v)", m, w))
			} else {
				break
			}
		}
	}
	fmt.Println(strings.Trim(fmt.Sprint(visitedList), "[]"))
}

/*
name   :05. Challenge the Wedding
input  :2 2 6
output :(1 <-> 1) (1 <-> 2) (2 <-> 1) (2 <-> 2)
*/
